mantidproject/mantid
# Fix template usage

The template function `getPropertyFromRun` contained in `LoadHelper.cpp` could not be used. Instead, it was copied to other files. 
- moved the template implementation to the header
- the MatrixWorkspace is not const anymore for more flexible usage that I need for the loader
- the include MatrixWorkspace.h must move to the header file as well 

**To test:**

Test possibility 1:
- checkout the different branch 18235_update_load_ill_reflectometry, where I used the template (test and source).
- make sure everything compiles (unit tests will pass for 80% due to current development)

Test possibility 2:
choose a source of a loader, then use the template for example like
Mantid::DataHandling::LoadHelper loadHelper;
or
#include "MantidDataHandling/LoadHelper.h"
LoadHelper loadHelper;
then
type sample_angle = loadHelper.getPropertyFromRun<type>(ws, "your_log_entry");
where ws is a MatrixWorkspace, the type should be the expected type of your_log_entry and your_log_entry should name a sample log entry.

Fixes #18748.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
